src/Provider/BaseVideoProvider.php

```diff
@@ -40,19 +42,54 @@ abstract class BaseVideoProvider extends BaseProvider
     protected $metadata;
 
     /**
-     * @param string $name
+     * @var ClientInterface|null
      */
-    public function __construct($name, Filesystem $filesystem, CDNInterface $cdn, GeneratorInterface $pathGenerator, ThumbnailInterface $thumbnail, Browser $browser, ?MetadataBuilderInterface $metadata = null)
-    {
+    private $client;
+
+    /**
+     * @var RequestFactoryInterface|null
+     */
+    private $requestFactory;
+
+    /**
+     * @param string                  $name
+     * @param ClientInterface|Browser $client
+     */
+    public function __construct(
+        $name,
+        Filesystem $filesystem,
+        CDNInterface $cdn,
+        GeneratorInterface $pathGenerator,
+        ThumbnailInterface $thumbnail,
+        object $client,
+        ?MetadataBuilderInterface $metadata = null,
+        ?RequestFactoryInterface $requestFactory = null
+    ) {
         parent::__construct($name, $filesystem, $cdn, $pathGenerator, $thumbnail);
 
         // NEXT_MAJOR: remove this check!
         if (!method_exists($this, 'getReferenceUrl')) {
             @trigger_error('The method "getReferenceUrl" is required with the next major release.', E_USER_DEPRECATED);
```

Comment:
I know this line is not changed in this PR, but if is possible, the message should be updated in order to be explicit about the version where the deprecation was added.